detection_rules/semver.py

```diff
@@ -12,7 +12,7 @@ def __new__(cls, version):
         if not isinstance(version, (int, list, tuple)):
             version = tuple(int(a) if a.isdigit() else a for a in re.split(r'[.-]', version))
 
-        return tuple.__new__(cls, version)
+        return tuple.__new__(cls, (version,) if isinstance(version, int) else version)
```

Comment:
if you just have a single integer there's no need to create a Version

Reply:
In its current state, an int will cause failures, so we have to perform a check anyway (and either pass the int through or cast it as a regular tuple). It seems useful to have the methods and be able to handle versions consistently by leveraging `Version`

Reply:
my thought is that if you already have an int, then converting it to a Version doesn't add any value

Reply:
I agree, so my assumption is anyone passing an int to `Version` either may not know the type of the passed version and want's consistency, or they want to take advantage of the methods.

I'm cool either way 🤷

Reply:
let's leave integers as is, and not convert to Version. you could do `Version((myint, ))` if you really want